src/js/preshuffle.s

```diff
@@ -756,17 +756,225 @@ MaybeSetCheckpoint:
 MaybeSetCheckpointActual:
 
 
+.segment "12", "13", "fe", "ff"
+;.bank $27000 $8000:$4000
+
+;; Overwrite the StatusBarDataTable with the new UI tile layout
+.org $badb
+  .byte $80,$20,$20,$20 ; Lf _ _ _
+.org $baee
+  .byte $83,$00,$00,$20 ; Lv 0 0 _
+.org $baf2
+  .byte $9c             ;  $
+
+.org $bafb
+  ;; on this row we shifted most everything left by three tiles
+  .byte $81,$8f,$8f,$8f ; Pw > > >
+  .byte $8f,$8f,$8f,$8f ;  > > > >
+  .byte $93,$95,$94     ; (1) lit up
+  .byte $90,$91,$92     ; (_) not lit up
+  .byte $90,$91,$92     ; (_) not lit up
+  .byte $20,$20         ;  _ _
+  .byte $84,$20,$20,$20 ; Dl _ _ _
+  .byte $86,$20,$20,$20 ; Ex _ _ _
+  .byte $20,$20,$1f     ;  _ _ |
+
+.org $bb1b
+  .byte $20             ;  _
+.org $bb1f ; clear out the experience count thats there initially
+  .byte $20,$20,$20,$20 ;  _ _ _ _
+  .byte $20,$20,$20,$20 ;  _ _ _ _
+  .byte $20,$20,$20,$20 ;  _ _ _ _
+  .byte $20,$20,$20,$20 ;  _ _ _ _
+  .byte $85,$20,$20,$20 ; Mp _ _ _
+  .byte $9d,$20,$20,$20 ;  / _ _ _
+
+
+
 .segment "1a", "1b", "fe", "ff"
 ;.bank $34000 $8000:$4000
 
 ;;; Numeric displays
 .org $8ee9  ; 06 - was LV(menu) but now it's difficulty
   .word (Difficulty)
-  .byte $3c,$2b,$03,$00 ; display right of lvl
+  .byte $56,$2b,$03,$00 ; display left of exp
 .org $8f19  ; 0e - was unused, now it's LV(menu)
   .word (PlayerLevel)
   .byte $29,$29,$03,$00 ; copied from $34ee9
 
+;; Move the position of the main UI number elements
+.org $8ec7 ; Lv
+  .byte $36,$2b
+.org $8ecd ; Money
+  .byte $3a,$2b
+.org $8ed3 ; Experience
+  .byte $5a,$2b
+; Experience to next level is now unused,
+; if you want to use it for something, make sure the drawing isn't skipped
+; .org $8ed9 ; Experience Cap
+;   .byte $5a,$2b
+
+.pushseg "13", "fe", "ff"
+;; Prevent EXP Left from being drawn
+;InitializeStatusBarNametable
+.org $bace
+  jsr DisplayNumberInternalButSkipExpToNextLevel
+.reloc
+DisplayNumberInternalButSkipExpToNextLevel:
+  cmp #$03
+  beq +
+  jmp DisplayNumberInternal
++ rts
+.popseg
+
+;;; HP / Force bar display
+.org $8d1f
+  lda #$23; Move the player HP bar left 3 places
+.org $8db2
+  lda #$43; Move the player force bar left 3 places
+
+;;; ----------------------------------------------------
+;;; KillObject
+;;; Recalculate PlayerEXP to count down from max instead of up
+
+.org $9152
+  ; Instead of calling AwardExperience immediately, just store the obj offset
+  ; and push it onto the stack for use later. The code needs to check for an
+  ; object replacement and we don't wanna clobber that
+  jmp PushObjOffsetToStack
+.assert * <= $9155
+
+.org $9155
+CheckForObjectReplacementOrDeathAnimation:
+
+.reloc
+PushObjOffsetToStack:
+  tya
+  pha
+  ; Return back to where we started without touching the stack
+  jmp CheckForObjectReplacementOrDeathAnimation
```

Comment:
Yeah the original reason i wrote it like this was because I wasn't planning on editing the part of AwardExperiencePoints because it does `lda ObjectExperiencePoints,y` in two different spots, I figured it'd be easier to just store the offset, but I think you have a point that it'd be better to remove those and instead read from a tmp value location instead. 

So originally it was `lda ObjectExperiencePoints,y` `pha` which didn't fit, so i moved it to this weird `jmp` to and back, but i changed it later when i figured i could just store `y` directly.